File: python3/fast_pow.py
def POW(a, b):
    res = 1

    while b > 0:
        if b % 2 != 0:
            res *= a
        a *= a
        b //= 2

    return res
  
  def POW(a, b, c):
    res = 1

    while b > 0:
        if b % 2 != 0:
            res *= a
            res %= c
        a *= a
        a %= c
        b //= 2

    return res

File: node_js/fast_pow.js
let pow = function (a, b, mod) {
    let res = BigInt(1);

    while (b > 0)
    {
        if (b & BigInt(1)) res = (res * a) % mod;

        b >>= BigInt(1);
        a = (a * a) % mod;
    }

    return res;
};
